dask/dataframe/indexing.py

```diff
@@ -58,7 +58,13 @@ def __getitem__(self, key):
         if iindexer != slice(None):
             raise NotImplementedError(msg)
 
-        return self._iloc(iindexer, cindexer)
+        if len(set(self.obj.columns)) < len(self.obj.columns):
```

Comment:
```suggestion
        if not self.obj.columns.is_unique:
```

Probably faster than converting to sets, and is cached by pandas.